pkg/api/customization/namespace/link.go

```diff
@@ -0,0 +1,99 @@
+package namespace
+
+import (
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/config"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/printers"
+	"strings"
+)
+
+var ExportResourcePrefixMappings = map[string]string{
+	"pods":                   "api/v1",
+	"services":               "api/v1",
+	"replicationcontrollers": "api/v1",
+	"deployments":            "apis/extensions/v1beta1",
+	"daemonsets":             "apis/extensions/v1beta1",
+	"replicasets":            "apis/extensions/v1beta1",
+	"statefulsets":           "apis/apps/v1beta1",
+	"jobs":                   "apis/batch/v1",
+	"cronjobs":               "apis/batch/v1beta1",
+}
+
+var ExportPrinters = map[string]printers.ResourcePrinter{
+	"json": &printers.JSONPrinter{},
+	"yaml": &printers.YAMLPrinter{},
+}
+
+func NewLinkHandler(next types.RequestHandler, client rest.Interface, management *config.ScaledContext) types.RequestHandler {
+	lh := &yamlLinkHandler{
+		next:       next,
+		client:     client,
+		management: management,
+	}
+
+	return lh.LinkHandler
+}
+
+type yamlLinkHandler struct {
+	next       types.RequestHandler
+	client     rest.Interface
+	management *config.ScaledContext
+}
+
+func (s *yamlLinkHandler) callNext(apiContext *types.APIContext, next types.RequestHandler) error {
+	if s.next != nil {
+		return s.next(apiContext, next)
+	} else if next != nil {
+		return next(apiContext, nil)
+	}
+
+	return httperror.NewAPIError(httperror.NotFound, "link not found")
+}
+
+func (s *yamlLinkHandler) LinkHandler(apiContext *types.APIContext, next types.RequestHandler) error {
+	if apiContext.Link != "yaml" {
+		return s.callNext(apiContext, next)
+	}
+
+	ns := apiContext.ID
+	result := &unstructured.UnstructuredList{}
+	result.SetAPIVersion("v1")
+	result.SetKind("List")
+
+	for kind, prefix := range ExportResourcePrefixMappings {
+
+		req := s.client.Get().Prefix(prefix).Namespace(ns).Resource(kind)
```

Comment:
This completely by passes authorization.  This client is running as a service account with all privileges.  You need to set the imperationation headers as the current user and group.  Refer to https://github.com/rancher/norman/blob/master/store/proxy/proxy_store.go#L117.

Also you need to handle a 403 response from the client and ignore resource that aren't authorized to that user